training/src/main/java/com/wolox/training/models/Book.java

```diff
@@ -54,4 +56,58 @@
 
   @ManyToMany(mappedBy = "books")
   private List<User> user;
+
+  public void setGenre(String genre) {
+    Preconditions.checkNotNull(genre,
+        "the genre " + AppConstants.PRECONDITION_BOOK_MESSAGE_NULL);
```

Comment:
You can use Preconditions.checkArgument with the Strings instead of checkNotNull, so I think inside checkArgument you can use Strings.isNullOrEmpty to check both conditions